packages/react-utils/src/components/KeyValuePairs/index.tsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import { Typography } from 'antd';
+import './index.css';
+
+const { Text } = Typography;
+
+type KeyValuePairs = Record<string, string | number | boolean | JSX.Element>;
+
+export const KeyValueGrid = (props: KeyValuePairs) => {
+  return (
+    <dl className="keyValueGrid">
+      {Object.entries(props).map(([key, value]) => {
+        return (
+          <div key={key} className="keyValueGrid-pair">
+            <dt className="keyValueGrid-pair__label">{key}</dt>
+            <dd className="keyValueGrid-pair__value">{value}</dd>
+          </div>
+        );
+      })}
+    </dl>
+  );
+};
+
+/**
+ * Use for single key value pair
+ *
+ * @param props - key value pair map
+ */
+export const SingleKeyNestedValue = (props: KeyValuePairs) => {
+  const firstPair = Object.entries(props)[0];
```

Comment:
wouldn't it be better to pass the first array item at the source and make this component agnostic?

Reply:
The props is an object, I do not yet know of a way to define an interface that limits an object to one arbitrary property:value mapping. The index is an additional sanity-ish kind of guarantee should someone pass more than one property in the props object